ote/src/cljs/ote/app/controller/transit_visualization.cljs

```diff
@@ -470,6 +446,34 @@
                                                              earlier-date
                                                              later-date)))))))
 
+(define-event LoadServiceChangesForDateResponse [response detection-date]
+  {}
+  (let [date-filter (if (= "now" (:scope app))
+                      (time/now-iso-date-str)
+                      detection-date)
+        route-hash (js/decodeURIComponent (get-in app [:params :route]))
+        changes (future-changes date-filter (:route-changes response))
+        route (some
```

Comment:
Should `some` call be wrapped within 
`when-let [route-hash (js/decodeURIComponent (get-in app [:params :route])]` 
to avoid iterating through `changes` when `route-hash`doesn't exist.